isovar/locus_reads.py

```diff
@@ -29,17 +28,34 @@
 )
 from .common import list_to_string
 from .dataframe_builder import DataFrameBuilder
-
+from .value_object import ValueObject
 
 logger = logging.getLogger(__name__)
 
-class LocusRead(namedtuple("LocusRead", [
+class LocusRead(ValueObject):
+    __slots__ = [
         "name",
         "sequence",
         "reference_positions",
         "quality_scores",
         "base0_read_position_before_variant",
-        "base0_read_position_after_variant"])):
+        "base0_read_position_after_variant"
+    ]
+
+    def __init__(
```

Comment:
> We haven't done benchmarking to know that the increased efficiency of field access/memory footprint/etc. 

Fair point; if we don't know that a special object is necessary, it's a complication. 

What I don't mind is automating away the __init__ self.arg assignment boilerplate, if after all these classes are just fancy namedtuples, using that facility of Python seems reasonable?